tutorials/integration.qmd

```diff
@@ -0,0 +1,171 @@
+---
+title: Integration
+---
+
+```{julia}
+#| echo: false
+#| code-fold: true
+#| output: false
+using Pkg;
+cd(@__DIR__)
+Pkg.activate("."); # for reproducibility use the local tutorial environment.
+Pkg.develop(PackageSpec(; path=(@__DIR__) * "/../"))
+using Markdown
+```
+
+This part of documentation covers integration of scalar functions defined on manifolds.
+The basic concepts are derived from geometric measure theory.
+In principle, there are many ways in which a manifold can be equipped with a measure that can be later used to define an integral.
+One of the most popular ways is based on pushing the Lebesgue measure on a tangent space through the exponential map.
+Any other suitable atlas could be used, not just the one defined by normal coordinates, though each one requires different volume density corrections due to the Jacobian determinant of the pushforward.
+`Manifolds.jl` provides the function ``[`volume_density`](@ref)``{=commonmark} that calculates that quantity, denoted $\theta_p(X)$.
+See for example [^leBrigantPuechmorel2019], Definition 11, for a precise description using Jacobi fields.
+
+While many sources define volume density as a function of two points, `Manifolds.jl` decided to use the more general point-tangent vector formulation. The two-points variant can be implemented as
+```{julia}
+using Manifolds
+volume_density_two_points(M::AbstractManifold, p, q) = volume_density(M, p, log(M, p, q))
+```
+
+The simplest way to of integrating a function on a compact manifold is through a Monte Carlo integrator.
+A simple variant can be implemented as follows (assuming uniform distribution of `rand`):
+
+```{julia}
+using LinearAlgebra, Distributions, SpecialFunctions
+function simple_mc_integrate(M::AbstractManifold, f; N::Int = 1000)
+    V = manifold_volume(M)
+    sum = 0.0
+    q = rand(M)
+    for i in 1:N
+        sum += f(M, q)
+        rand!(M, q)
+    end
+    return V * sum/N
+end
+```
+
+``[`manifold_volume`](@ref)``{=commonmark} and ``[`volume_density`](@ref)``{=commonmark} are closely related to each other, though very few sources explore this connection, and some even claiming a certain level of arbitrariness in defining `manifold_volume`.
```

Comment:
This to mea reads like a very abrupt change of topic? Maybe introduce this a bit, maybe as, “We used the function ... which is related to”?

```suggestion
[`manifold_volume`](@ref) and [`volume_density`](@ref) are closely related to each other, though very few sources explore this connection, and some even claiming a certain level of arbitrariness in defining [`manifold_volume`](@ref).
```
the commonmark (and = to keep the code unchanged) is only necessary of there are spaces in the `(@ref)` part, since otherwise these are escaped in an HTML manner (i.e. to `%20`).

Reply:
Good point, I've moved that part to a separate section at the end. I've also removed the `commonmark` thing where it looked unnecessary.